hack/generate.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Ignore the rule that says we should always quote variables, because
+# in this script we *do* want globbing.
+# shellcheck disable=SC2086
+
+set -eux
+
+IS_CONTAINER=${IS_CONTAINER:-false}
+ARTIFACTS=${ARTIFACTS:-/tmp}
+CONTAINER_RUNTIME="${CONTAINER_RUNTIME:-podman}"
+
+if [ "${IS_CONTAINER}" != "false" ]; then
+  eval "$(go env)"
+  cd "${GOPATH}"/src/github.com/metal3-io/baremetal-operator
+  export XDG_CACHE_HOME="/tmp/.cache"
+
+  operator-sdk version
+
+  INPUT_FILES="deploy/crds/*.yaml pkg/apis/metal3/v1alpha1/zz_generated.*.go"
+  cksum $INPUT_FILES > "$ARTIFACTS/lint.cksums.before"
+  export VERBOSE="--verbose"
+  make generate
+  cksum $INPUT_FILES > "$ARTIFACTS/lint.cksums.after"
```

Comment:
One downside of only storing checksums is that if you *had* made local changes then `make generate-check` or `make lint` will make them go bye-bye, which could be a bit surprising. That was the reason for my hacky gymnastics in the Makefile previously.

Reply:
What is the use case for editing the YAML by hand?

Reply:
Nothing valid, but if you didn't realise you weren't supposed to and only found out by using the linter then the info you wanted to copy-and-paste into the right place was just permanently deleted.
It also means `make lint` is not idempotent; running it a second time will pass. You might end up pushing a commit that actually fails if you don't notice the uncommitted changes (although at least there's no data loss in that case).